src/main/java/bio/terra/service/snapshot/SnapshotService.java

```diff
@@ -195,6 +195,11 @@ public String createSnapshot(
     UUID snapshotId = UUID.randomUUID();
     String description =
         "Create snapshot %s with ID %s".formatted(snapshotRequestModel.getName(), snapshotId);
+    
+    snapshotRequestModel.setContents(
+        snapshotRequestModel.getContents().stream()
+            .map(model -> model.datasetName(dataset.getName()))
+            .toList());
```

Comment:
Would you mind adding a note here or updating the PR description? I admint it is a bit of an odd call so I want to help us remember why we needed this.

Reply:
Added a comment!